src/main/java/seedu/address/model/person/task/TaskDeadline.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.model.person.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Task's taskDeadline in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTaskDeadline(String)}
+ */
+public class TaskDeadline {
+    public static final String MESSAGE_CONSTRAINTS = "Task deadline should be in the format YYYY-MM-DD.";
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    public final LocalDate taskDeadline;
```

Comment:
Should taskDeadline be private?

Reply:
hmm for this i followed how all the other Person fields are defined, all of their values of their things are public

Reply:
Oh okay didn't realise that